onelogin/ruby-saml
# Backport multi-value attribute support to 0.8.x

## Status
**READY**

## Migrations
NO

## Description
Using the code from `v1.4.0` the feature to extract multi-value attributes has been back-ported to the 0.8.x branch.

## Steps to Test or Reproduce
1. Obtain a SAML Response with an `AttributeStatement` node containing attributes with multiple values, such as [this one](https://github.com/onelogin/ruby-saml/blob/010281f5fa4ffa2a718b45907d1c005a6f088564/test/responses/response_with_multiple_attribute_values.xml) and store it in a variable (Let's say `saml_response_doc`).
2. Instantiate a Response object.
```ruby 
response = OneLogin::RubySaml::Response.new(saml_response_doc)
```
3. Turn off `single_value_compatibility` in the `Attributes` class, and extract the value.
```ruby
OneLogin::RubySaml::Attributes.single_value_compatibility = false
roles = response.attributes[:role]
```
4. Verify that your extracted value is correct.

## Impacted Areas in Application
* SAML Response attributes parsing
